src/node_api.cc

```diff
@@ -34,17 +39,46 @@ v8::Maybe<bool> node_napi_env__::mark_arraybuffer_as_untransferable(
 }
 
 void node_napi_env__::CallFinalizer(napi_finalize cb, void* data, void* hint) {
-  // we need to keep the env live until the finalizer has been run
-  // EnvRefHolder provides an exception safe wrapper to Ref and then
-  // Unref once the lambda is freed
-  EnvRefHolder liveEnv(static_cast<napi_env>(this));
-  node_env()->SetImmediate(
```

Comment:
The `SetImmediate()` is to ensure that JS code is actually runnable in the target environment – where is this happening now?

Reply:
Actually, I find that we already have a outer SetImmediate in BufferFinalizer: https://github.com/nodejs/node/blob/master/src/node_api.cc#L65

Anyway, this is not related to the intention of this PR. I'm going to revert this.

Reply:
I merged the setimmediate in bufferfinalizer to this one since they are doing the same thing.